x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_sync_rules_table_saved_state.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { RULES_TABLE_STATE_STORAGE_KEY } from '../../../../../../common/constants';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { URL_PARAM_KEY } from '../../../../../common/hooks/use_url_state';
+import { useUpdateUrlParam } from '../../../../../common/utils/global_query_string';
+import { AllRulesTabs } from '../rules_table_toolbar';
+
+import { useRulesTableContext } from './rules_table_context';
+import type { RulesTableSavedState } from './rules_table_saved_state';
+
+export function useSyncRulesTableSavedState(activeTab: AllRulesTabs): void {
+  const { state } = useRulesTableContext();
+  const {
+    services: { storage },
+  } = useKibana();
+  const updateUrlParam = useUpdateUrlParam<RulesTableSavedState>(URL_PARAM_KEY.rulesTable);
+
+  useEffect(() => {
+    const savedState: RulesTableSavedState = {
+      filter: state.filterOptions,
+      inMemory: state.isInMemorySorting,
+      sort: state.sortingOptions,
+      page: state.pagination.page,
+      perPage: state.pagination.perPage,
+    };
+
+    if (activeTab === AllRulesTabs.monitoring) {
+      savedState.tab = activeTab;
+    }
+
+    updateUrlParam(savedState);
+    storage.set(RULES_TABLE_STATE_STORAGE_KEY, savedState);
```

Comment:
I have doubts about persisting the current table state in local storage. It provides quite a confusing user experience. For example, if I select table filter in one tab, then open another tab and navigate to the rules table, I'll see the state from the previous tab: 

https://user-images.githubusercontent.com/1938181/202178637-b13492bf-7238-4dc7-9244-2f3b5bce4a23.mov

This behavior is pretty uncommon and could be unexpected from a user standpoint.

On top of that, local storage has a long expiration time. That means if I filtered the table once and then returned to it one month later, I'll see it in the state I left it a month ago, which could be pretty stale. So users will likely have to reset the table filters to their initial state every time they open it. 

My suggestion here is to use either session storage for the table filters or a combination of URL + in-memory state, how it's done in other areas of security solution.

Reply:
++ I think using session storage might be a good idea. I'd suggest to test how it works in different browsers because I suspect sessions are not cleared immediately after a user closes their tabs. And I think session storage is a simpler and better persistence mechanism compared to an in-memory state via redux (in our case, of course).

Reply:
I've switched to the session storage and validated the scenarios in Chrome so it keeps a state copy per tab and doesn't restore it after the tab was closed.